app/helpers/sessions_helper.rb

```diff
@@ -4,21 +4,45 @@ def log_in user
   end
 
   def current_user
-    @current_user ||= User.find_by(id: session[:user_id])
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by id: user_id
+    elsif (user_id = cookies.signed[:user_id])
+      @current_user = user_from_remember_token(user_id)
+    end
+  end
+
+  def user_from_remember_token(user_id)
+    user = User.find_by id: user_id
+    if user&.authenticated?(cookies[:remember_token])
+      log_in user
+      user
+    end
```

Comment:
```suggestion
    user = User.find_by id: user_id
    return unless user&.authenticated?(cookies[:remember_token])
    
    log_in user
    user
```
E viết kiểu return if cho clean code nha